### PR TITLE
#163388256 Users should be able to comment on articles 

### PR DESCRIPTION
**Pull request by TooColline**
#### What does this PR do?
This PR Allows users to comment on articles and reply to comments

#### Description of Task to be completed?
- Allow users to comment on articles and replies
- Allow users to reply to comments and replies
- Allow users to get all comments and replies
- Allow users to get specific comments or replies
- Allow users to edit their own comments or replies
- Allow users to delete their own comments or replies

#### How should this be manually tested?
Run the server using ```python manage.py runserver```

**Go to Postman and make the following requests:**

Method | Endpoint | Functionality
--- | --- | ---
POST | `{{base-url}}/articles/<article-slug>/comments/` | Post a new comment
POST | `{{base-url}}/articles/<article-slug>/comments/<comment_id>/` | Post a new reply to a comment
GET | `{{base-url}}/articles/<article-slug>/comments/` | Retrieve all comments and replies
GET | `{{base-url}}/articles/<article-slug>/comments/<comment_id>` | Retrieve a specific comment or reply
PUT | `{{base-url}}/articles/<article-slug>/comments/<comment_id>/` | Update a specific comment or reply
DELETE | `{{base-url}}/articles/<article-slug>/comments/<comment_id>/` | Delete a specific comment or reply

#### Any background context you want to provide?
Comments will allow authors to get and give feedback on the articles
```{{base-url}}``` is configured to ```http://127.0.0.1:8000/api```

#### What are the relevant pivotal tracker stories?
[#163388256](https://www.pivotaltracker.com/n/projects/2240063)

#### Screenshots (if appropriate)
**Post Comment**
image

**Post a reply**
image

**Get all comments**
image

**Get specific comment**
image

**Update comment**
image

**Delete comment**
image

**Delete comment when unauthorized**
image

**Reply to a comment that does not exist**
image

**Post an empty comment**
image

#### Questions:
None

### CODE REVIEW

File: authors/apps/comments/views.py [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+
+class CommentDetailsAPIView(RetrieveUpdateDestroyAPIView, ListCreateAPIView):
+    """
+    Creates a view to Get, update, and delete a particular comment
+    """
+
+    permission_classes = (IsAuthenticatedOrReadOnly,)
+    serializer_class = CommentsSerializers
+
+    def get(self, request, id, *args, **kwargs):
```
Comment (sam-karis) [inline authors/apps/comments/views.py]: I suggest you do away with the `*args` argument is not used anywhere thus not necessary.

File: authors/apps/comments/views.py [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+            'author': request.user.id
+        }
+
+        serializer = self.serializer_class(
+            data=data)
+        serializer.is_valid(raise_exception=True)
+        serializer.save(article=article, author=request.user)
+        return Response(serializer.data, status=status.HTTP_201_CREATED)
+
+    def destroy(self, request, id, *args, **kwargs):
```
Comment (sam-karis) [inline authors/apps/comments/views.py]: The same as above`*args` is not necessary.

File: authors/apps/comments/views.py [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+        return Response(
+            {
+                "comment": {
+                    "message": "Comment was successfully deleted"
+                }
+            },
+            status=status.HTTP_200_OK
+        )
+
+    def update(self, request, id, *args, **kwargs):
```
Comment (sam-karis) [inline authors/apps/comments/views.py]: The same as above`*args` is not necessary.

**Review by sam-karis — CHANGES_REQUESTED** [review]
Good work @TooColline, implement the few changes above and the work will be ready for merging

**Review by sam-karis — APPROVED** [review]
LGTM

**Review by calebrotich — APPROVED** [review]
Awesome work @TooColline , works as expected...

**Review by bryan-munene — APPROVED** [review]
Looks Good

**Pull request merged**